Quarkus 3.22 - Gradle Unit Tests do not run in multi module setup when not all modules have a test

### Describe the bug

1. Run `./gradlew quarkusDev` 
2. Toggle test output 
3. run tests

### Expected behavior

Tests run

### Actual behavior

```
java.lang.RuntimeException: java.util.NoSuchElementException: No value present
        at io.quarkus.deployment.dev.testing.JunitTestRunner.prepare(JunitTestRunner.java:447)
        at io.quarkus.deployment.dev.testing.ModuleTestRunner.prepare(ModuleTestRunner.java:83)
        at io.quarkus.deployment.dev.testing.TestSupport.runInternal(TestSupport.java:430)
        at io.quarkus.deployment.dev.testing.TestSupport$3.run(TestSupport.java:389)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:143)
        at io.quarkus.deployment.dev.testing.JunitTestRunner.discoverTestClasses(JunitTestRunner.java:716)
        at io.quarkus.deployment.dev.testing.JunitTestRunner.prepare(JunitTestRunner.java:166)
        ... 4 more
```

### How to Reproduce?

https://github.com/kesslerd/quarkus-gradle-unit-test-bug

I have a gradle project with multiple modules. 
Not all modules contain a test folder
The setup works with 3.21.4



### Output of `uname -a` or `ver`

24.3.0 Darwin Kernel Version 24.3.0: Thu Jan  2 20:24:16 PST 2025; root:xnu-11215.81.4~3/RELEASE_ARM64_T6000 arm6

### Output of `java -version`

OpenJDK Runtime Environment Zulu21.40+17-CA (build 21.0.6+7-LTS)

### Quarkus version or git rev

3.22.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.14

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java
@@ -569,21 +569,23 @@ public class JunitTestRunner {
         //for now this is out of scope, we are just going to do annotation based discovery
         //we will need to fix this sooner rather than later though
 
+        if (moduleInfo.getTest().isEmpty()) {
+            return DiscoveryResult.EMPTY;
+        }
+
         //we also only run tests from the current module, which we can also revisit later
         Indexer indexer = new Indexer();
-        moduleInfo.getTest().ifPresent(test -> {
-            try (Stream<Path> files = Files.walk(Paths.get(test.getClassesPath()))) {
-                files.filter(s -> s.getFileName().toString().endsWith(".class")).forEach(s -> {
-                    try (InputStream in = Files.newInputStream(s)) {
-                        indexer.index(in);
-                    } catch (IOException e) {
-                        throw new RuntimeException(e);
-                    }
-                });
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        });
+        try (Stream<Path> files = Files.walk(Paths.get(moduleInfo.getTest().get().getClassesPath()))) {
+            files.filter(s -> s.getFileName().toString().endsWith(".class")).forEach(s -> {
+                try (InputStream in = Files.newInputStream(s)) {
+                    indexer.index(in);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
 
         Index index = indexer.complete();
         //we now have all the classes by name
@@ -699,6 +701,20 @@ public class JunitTestRunner {
             unitTestClasses.add(name);
         }
 
+        // if we didn't find any test classes, let's return early
+        // Make sure you also update the logic for the non-empty case above if you adjust this part
+        if (testType == TestType.ALL) {
+            if (unitTestClasses.isEmpty() && quarkusTestClasses.isEmpty()) {
+                return DiscoveryResult.EMPTY;
+            }
+        } else if (testType == TestType.UNIT) {
+            if (unitTestClasses.isEmpty()) {
+                return DiscoveryResult.EMPTY;
+            }
+        } else if (quarkusTestClasses.isEmpty()) {
+            return DiscoveryResult.EMPTY;
+        }
+
         List<Class<?>> itClasses = new ArrayList<>();
         List<Class<?>> utClasses = new ArrayList<>();
 
@@ -809,6 +825,7 @@ public class JunitTestRunner {
             }
         }
 
+        // Make sure you also update the logic for the empty case above if you adjust this part
         if (testType == TestType.ALL) {
             //run unit style tests first
             //before the quarkus tests have started
@@ -1270,6 +1287,8 @@ public class JunitTestRunner {
     }
 
     static class DiscoveryResult implements AutoCloseable {
+
+        private final static DiscoveryResult EMPTY = new DiscoveryResult(null, List.of());
 
         final QuarkusClassLoader classLoader;
         final List<Class<?>> testClasses;

